src/server/pkg/deploy/cmds/cmds.go

```diff
@@ -17,92 +17,115 @@ import (
 	"go.pedge.io/pkg/exec"
 )
 
+func maybeKcCreate(dryRun bool, manifest *bytes.Buffer) error {
+	if dryRun {
+		n, err := os.Stdout.Write(manifest.Bytes())
+		if err == nil && n < manifest.Len() {
+			return io.ErrShortWrite
+		}
+		return err
```

Comment:
Nice thorough error checking here but you I believe you can get the same effect with:

```
_, err := os.Stdout.Write(manifest.Bytes())
return err
```

Go specifies in the semantics of `io.Writer` that:

>  Write must return a non-nil error if it returns n < len(p).

https://golang.org/pkg/io/#Writer